ChatSecure/Classes/Model/Yap Storage/OTRXMPPRoomOccupant.swift

```diff
@@ -64,11 +65,25 @@ open class OTRXMPPRoomOccupant: OTRYapDatabaseObject, YapDatabaseRelationshipNod
     @objc open var affiliation:RoomOccupantAffiliation = .none
 
     /**When given by the server we get the room participants reall JID*/
-    @objc open var realJID:String?
+    @objc open var realJID:String? {
+        didSet {
+            if let realJid = self.realJID, let roomUniqueId = self.roomUniqueId {
+                OTRDatabaseManager.shared.readOnlyDatabaseConnection?.asyncRead({ (transaction) in
+                    if let room = OTRXMPPRoom.fetchObject(withUniqueID: roomUniqueId, transaction: transaction), let accountUniqueId = room.accountUniqueId {
+                        self.realBuddy = OTRBuddy.fetch(withUsername: realJid, withAccountUniqueId: accountUniqueId, transaction: transaction)
+                    }
+                })
+            }
+        }
+    }
+    @objc open var realBuddy:OTRBuddy?
```

Comment:
Instead of doing database transactions in didSet and then assigning a whole database object, I think it would be better to:

1. Fetch the corresponding OTRBuddy in the YapStorage class
2. Just store the OTRBuddy.uniqueId on `OTRXMPPRoomOccupant.buddyUniqueId` as a `String?`.
3. Add an accessor function `func buddy(with: YapDatabaseReadTransaction) -> OTRXMPPBuddy?` that does a simple lookup.